aurora/aurora-war/src/main/java/org/silverpeas/looks/aurora/AuroraSpaceHomePage.java

```diff
@@ -0,0 +1,372 @@
+    }
+    if (nbPublicationsToDisplay > 0) {
+      return look.getLastUpdatedPublicationsSince(getSpace().getId(), 0, nbPublicationsToDisplay);
+    }
+    return Collections.emptyList();
+  }
+
+  public List<PublicationDetail> getNews() {
+    boolean displayNews = isEnabled("space.homepage.news", "displayNews");
+    if (displayNews) {
+      return look.getNews(getSpace().getId());
+    }
+    return Collections.emptyList();
+  }
+
+  public boolean isTaxonomyEnabled() {
+    return getFieldBooleanValue("displayTaxonomy");
+  }
+
+  public List<Space> getSubSpaces() {
+    boolean display = isEnabled("space.homepage.subspaces", "displaySubspaces");
+    if (display) {
+      OrganizationController oc = OrganizationController.get();
+      // get allowed subspaces
+      String[] subspaceIds = oc.getAllowedSubSpaceIds(look.getUserId(), getSpace().getId());
+      List<Space> subspaces = new ArrayList<>();
+      for (String subspaceId : subspaceIds) {
+        subspaces.add(new Space(oc.getSpaceInstLightById(subspaceId), userLanguage));
+      }
+      return subspaces;
+    }
+    return Collections.emptyList();
+  }
+
+  public List<App> getApps() {
+    boolean display = isEnabled("space.homepage.apps", "displayApps");
+    if (display) {
+      List<App> apps = new ArrayList<>();
+      List<ComponentInstLight> components = getAllowedComponents(true, null);
+      for (ComponentInstLight component : components) {
+        apps.add(new App(component, userLanguage));
+      }
+      return apps;
+    }
+    return Collections.emptyList();
+  }
+
+  public List<UserDetail> getAdmins() {
+    boolean displayAdmins = isEnabled("space.homepage.admins", "displayAdmins");
+    if (displayAdmins) {
+      return look.getSpaceAdmins(getSpace().getId());
+    }
+    return Collections.emptyList();
+  }
+
+  public NextEvents getNextEvents() {
+    boolean displayEvents = isEnabled("space.homepage.events", "displayEvents");
+    if (displayEvents) {
+      // get allowed apps
+      List<String> allowedAppIds = new ArrayList<>();
+      List<ComponentInstLight> components = getAllowedComponents(false, "almanach");
+      for (ComponentInstLight component : components) {
+         allowedAppIds.add(component.getId());
+      }
+      return look.getNextEvents(allowedAppIds);
+    }
+    return null;
+  }
+
+  private List<ComponentInstLight> getAllowedComponents(boolean visibleOnly, String name) {
+    OrganizationController oc = OrganizationController.get();
+    String[] appIds = oc.getAvailCompoIdsAtRoot(getSpace().getId(), look.getUserId());
+    List<ComponentInstLight> components = new ArrayList<>();
+    for (String appId : appIds) {
+      ComponentInstLight component = oc.getComponentInstLight(appId);
+      if (!visibleOnly || (visibleOnly && !component.isHidden())) {
+        if (StringUtil.isNotDefined(name) || (StringUtil.isDefined(name) && component.getName().equals(name))) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1066)